scipy/linalg/tests/test_lapack.py

```diff
@@ -1654,3 +1655,80 @@ def test_geqrfp_lwork(dtype, shape):
     m, n = shape
     lwork, info = geqrfp_lwork(m=m, n=n)
     assert_equal(info, 0)
+
+
+def generate_random_dtype_array(shape, dtype):
+    # generates a random matrix of desired data type of shape
+    if type(shape) != tuple:
+        # in case shape is not passed in as a tuple
+        shape = (shape,)
+    if dtype in COMPLEX_DTYPES:
+        return (np.random.rand(*shape)
+                + np.random.rand(*shape)*1.0j).astype(dtype)
+    return np.random.rand(*shape).astype(dtype)
+
+
+@pytest.mark.parametrize('dtype', DTYPES)
+@pytest.mark.parametrize('matrix_size', [(6, 7), (7, 6), (6, 6)])
+def test_geqrfp(dtype, matrix_size):
+    # set test tolerance appropriate for dtype
+
+    np.random.seed(42)
+    rtol = 250*np.finfo(dtype).eps
+    atol = np.finfo(dtype).eps
```

Comment:
This might be too stringent. I think (at)ilayn suggested a tolerance on the gttrf/gttrs issue or PR - please take a look for that.

Reply:
Got it, I saw that. I’ll update this